kroxylicious-filters/kroxylicious-sasl-inspection/src/main/java/io/kroxylicious/filters/sasl/inspection/SaslObserver.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filters.sasl.inspection;
+
+import org.apache.kafka.common.errors.AuthenticationException;
+
+/**
+ * A Sasl observer merely watches a SASL negotiation between client and server extracting the
+ * authorization id.  It is also responsible for signalling when the SASL negotiation
+ * is finished.
+ * <br/>
+ * Unlike a {@link javax.security.sasl.SaslServer}, it does not if the authentication was successful or not.
+ */
+public interface SaslObserver {
+
+    /**
+     * IANA registered SASL mechanism name.
```

Comment:
No harm in linking to https://www.iana.org/assignments/sasl-mechanisms/sasl-mechanisms.xhtml